src/main/java/net/oauth/jsontoken/exceptions/InvalidJsonTokenException.java

```diff
@@ -0,0 +1,29 @@
+package net.oauth.jsontoken.exceptions;
+
+public final class InvalidJsonTokenException extends Exception {
+  private final ErrorCode errorCode;
+
+  public InvalidJsonTokenException(ErrorCode errorCode) {
+    this.errorCode = errorCode;
+  }
+
+  public InvalidJsonTokenException(ErrorCode errorCode, String message) {
+    super(message);
+    this.errorCode = errorCode;
+  }
+
+  public InvalidJsonTokenException(ErrorCode errorCode, String message, Throwable cause) {
+    super(message, cause);
+    this.errorCode = errorCode;
+  }
+
+  public InvalidJsonTokenException(ErrorCode errorCode, Throwable cause) {
+    super(cause);
+    this.errorCode = errorCode;
+  }
```

Comment:
Where are these constructors used?

Reply:
The first constructor will be used for when InvalidJsonTokenException is immediately thrown with a message (eg. no verifiers). The third constructor will be used when we rethrow exceptions, in case we may want to keep the underlying exception thrown. As for the second constructor, I don't think there are any use cases so far. 

I included all of these constructors to allow for flexibility when constructing an InvalidJsonTokenException. My reasoning was that InvalidJsonTokenException was just another exception with a specialized error code, so the underlying constructors will be identical (with the addition of a error code). Should the unused one still be kept?